docusaurus/docs/publish-a-plugin/publishing-best-practices.md

```diff
@@ -0,0 +1,140 @@
+
+**[Screenshots](../reference/plugin-json#info)**
+
+`info.screenshots`
+
+The screenshots field should be used to provide an array of screenshot images that will be displayed within the plugin catalog. This is great for providing users with a visual representation of your plugin and can help them establish whether or not this plugin solves their problem before they even install it. Be sure to provide screenshots of your plugin in action, highlighting the standout features.
+
+:::note
+
+Ensure your screenshots are a suitable resolution and file type (e.g. png, jpeg, or gif).
+
+:::
+
+**[Grafana version compatibility](../reference/plugin-json#dependencies)**
+
+`dependencies.grafanaDependency`
+
+Ensure your plugin specifies the minimum Grafana version it is compatible with. This guarantees that users running different versions of Grafana know whether your plugin will work for them. Be sure to [run end-to-end tests](../e2e-test-a-plugin/introduction.md) to confirm compatibility with releases you support.
+
+## Creating a comprehensive README
+
+Your plugin's README file serves as both a first impression and a detailed guide for your users. Think of it as a combination of a storefront advertisement and an instruction manual - showing what your plugin can do, how to install it, and how users can make the most of it within their Grafana instances.
+
+To assist developers in crafting a high-quality README, we provide a [README template](https://raw.githubusercontent.com/grafana/plugin-tools/main/packages/create-plugin/templates/common/src/README.md) as part of the plugin structure generated by the `create-plugin` tool. This template ensures you cover the essential components while giving you flexibility to add more specific details.
+
+Beyond a basic overview of your plugin, its use cases, and requirements, there are additional elements you should consider including to help users understand the value and functionality of your plugin:
+
+- **Screenshots or screen captures:** Visual aids often communicate better than text alone. Including screenshots or even video demonstrations allows users to quickly grasp the plugin’s capabilities and setup process, giving them confidence to use it effectively.
+- **Dynamic badges:** Badges provide quick, at-a-glance information about your plugin, such as the latest release version or whether it has passed security and code checks. Tools like [shields.io](https://shields.io/) can be used with the Grafana.com API to automatically update these badges whenever you publish a new version, adding transparency and trustworthiness to your plugin.
+- **Contribution guidance:** Maintaining a plugin can be demanding, especially for individual developers. Clearly outlining how users can provide feedback, report bugs, or contribute code improvements helps foster community involvement, making it easier to maintain and improve your plugin over time.
```

Comment:
We should make sure that people don't confuse `contribute code improvements` with instructions on how to clone, build and develop the plugin. We reject submissions of readme files that contain "development slang".

Reply:
@academo any particular reason we reject plugin submissions for that reason? We've seen recently a case where a maintainer of a large plugin wanted to hand over the reigns but did not have any contributors. Putting contribution information in a README is fairly common practice. If not in the README, where should plugin developers put this information?

Reply:
It's mainly as the plugin readme is primarily there for users and developers who may contribute are a small % of that group. Agree that advertising how to contribute is important, but it should be a signpost in the readme with details in a contributing.md as you say.